cdap-storage-spi/src/main/java/io/cdap/cdap/spi/data/StructuredTable.java

```diff
@@ -169,6 +169,28 @@ default CloseableIterator<StructuredRow> scan(Range keyRange, int limit, Field<?
     throw new UnsupportedOperationException("No supported implementation.");
   }
 
+  /**
+   * Read a set of rows from the table matching the index and the key range.
+   * The rows returned will be sorted on the primary key order.
+   *
+   * @param keyRange key range for the scan
+   * @param limit maximum number of rows to return
+   * @param filterIndex the index to filter upon
+   * @param sortOrder defined primary key sort order. Note that the comparator used is specific to the underlying
+   *                  store and is not nessesarily lexographic.
+   * @return a {@link CloseableIterator} of rows
+   * @throws InvalidFieldException if the field is not part of the table schema, or is not an indexed column,
+   *                               or the type does not match the schema
+   * @throws IOException if there is an error scanning the table
+   */
+  default CloseableIterator<StructuredRow> scan(Range keyRange, int limit, Field<?> filterIndex, SortOrder sortOrder)
```

Comment:
Add a unit test on StructuredTable Level (e.g. StructuredTableTest) including one with filter and DESC sort order.